docarray/computation/abstract_comp_backend.py

```diff
@@ -0,0 +1,127 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, List, Optional, Tuple, Union
+
+if TYPE_CHECKING:
+    from docarray.typing.tensor.tensor import Tensor
+
+
+class AbstractComputationalBackend(ABC):
+    """
+    Abstract base class for computational backends.
+    Every supported tensor/ML framework (numpy, torch etc.) should define its own
+    computational backend exposing common functionality expressed in that framework.
+    That way, DocArray can leverage native implementations from all frameworks.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def stack(
+        tensors: Union[List['Tensor'], Tuple['Tensor']], dim: int = 0
+    ) -> 'Tensor':
+        """
+        Stack a list of tensors along a new axis.
+        """
+        ...
+
+    class Retrieval(ABC):
```

Comment:
why is this inside the other class ?

Reply:
just for further structuring/namespacing things. so all the metrics can go in one place, all the retrieval related stuff in another place, etc